src/main/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateExecution.java

```diff
@@ -91,4 +126,42 @@ private String computePathOnDisk(String physicalPathOnDisk) {
         FilePath diskFilePath = new FilePath(new File(physicalPathOnDisk));
         return new FilePath(diskFilePath, run.getParent().getFullName() + "/" + run.getNumber()).getRemote();
     }
+
+    /**
+     * Find the upstream job that matches the given upstream name.
+     *
+     * @param upstreamJob the upstream job name
+     * @return the upstream job with the given name if any, {@code null} otherwise
+     */
+    @CheckForNull
+    private Item findUpstreamJob(String upstreamJob) {
```

Comment:
This is the way how finding the upstream job works:
- the upstream job must be in the same Folder with the downstream job
- the user must pass only the name of the upstream job

I'm not quite sure how finding of the upstream job should work.
@oleg-nenashev you've mentioned about passing the full name when calling `exwsAllocate upstream: 'upstream-full-name'`. In this case, I would have to use `Jenkins.getInstance().getItemByFullName(upstreamJob)`? And this means that the upstream and downstream jobs don't have to be in the same Folder?

I'm saying upstream/downstream in the external workspace manager plugin context, and not how they are defined in Jenkins.